ocaml/typing/parmatch.ml

```diff
@@ -1988,30 +1988,40 @@ let do_check_partial ~pred loc casel pss = match pss with
       exhaust None pss (List.length ps)
       |> Seq.filter_map (typecheck ~pred) in
     match counter_examples () with
-    | Seq.Nil -> Total
+    | Seq.Nil ->
+        (match warn_if with
+          | Total ->
+              if Warnings.is_active Total_match_in_pattern_guard
+              then Location.prerr_warning loc Total_match_in_pattern_guard
+          | Partial -> ());
```

Comment:
Nit: it's often a good idea to put short cases first. Could you move up the `Partial` case here (and the `Total` case below)? Feel free to resolve after addressing.